addon/wait-for-coroutine.ts

```diff
@@ -0,0 +1,74 @@
+import TestWaiter from './test-waiter';
+
+const COROUTINE_WAITER = new TestWaiter<Generator>('coroutine-waiter');
+
+/**
+ * A convenient utility function to simplify waiting for a coroutine in a
+ * context such as ember-concurrency. Can be used in both decorator and
+ * function form.
```

Comment:
Let's beef this up to be more clear about what waiting means in this context. I'd like to see a dumbed down "for example, in the coroutine below, the decorator will beginAsync on line X and endAsync on line Y".

Reply:
👍